clippy_lints/src/literal_string_with_formatting_arg.rs

```diff
@@ -0,0 +1,113 @@
+use rustc_ast::ast::{Expr, ExprKind};
+use rustc_ast::token::LitKind;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_parse_format::{ParseMode, Parser, Piece};
+use rustc_session::declare_lint_pass;
+use rustc_span::BytePos;
+
+use clippy_utils::diagnostics::span_lint;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks if string literals have formatting arguments outside of macros
+    /// using them (like `format!`).
+    ///
+    /// ### Why is this bad?
+    /// It will likely not generate the expected content.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// let x: Option<usize> = None;
+    /// let y = "hello";
+    /// x.expect("{y:?}");
+    /// ```
+    /// Use instead:
+    /// ```no_run
+    /// let x: Option<usize> = None;
+    /// let y = "hello";
+    /// x.expect(&format!("{y:?}"));
+    /// ```
+    #[clippy::version = "1.83.0"]
+    pub LITERAL_STRING_WITH_FORMATTING_ARG,
```

Comment:
I guess this should be plural, according to lint naming guidelines

```suggestion
    pub LITERAL_STRINGS_WITH_FORMATTING_ARG,
```